docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,90 @@
+* ``CAS`` (Compare and Swap)
+* ``FetchADD``
+* ``SWAP``
 
-The PCIe 3.0 atomic operations feature allows atomic transactions to be requested by, routed through
-and completed by PCIe components. Routing and completion does not require software support.
-Component support for each is detectable via the Device Capabilities 2 (DevCap2) register. Upstream
-bridges need to have atomic operations routing enabled or the atomic operations will fail even though
-PCIe endpoint and PCIe I/O devices has the capability to atomic operations.
-
-To do atomic operations routing capability between two or more Root Ports, each associated Root Port
-must indicate that capability via the atomic operations routing supported bit in the DevCap2 register.
-
-If your system has a PCIe Express Switch it needs to support atomic operations routing. Atomic
-operations requests are permitted only if a component's ``DEVCTL2.ATOMICOP_REQUESTER_ENABLE``
-field is set. These requests can only be serviced if the upstream components support atomic operation
-completion and/or routing to a component which does. Atomic operations routing support=1, routing
-is supported; atomic operations routing support=0, routing is not supported.
-
-An atomic operation is a non-posted transaction supporting 32-bit and 64-bit address formats, there
-must be a response for Completion containing the result of the operation. Errors associated with the
-operation (uncorrectable error accessing the target location or carrying out the atomic operation) are
-signaled to the requester by setting the Completion Status field in the completion descriptor, they are
-set to to Completer Abort (CA) or Unsupported Request (UR).
-
-To understand more about how PCIe atomic operations work, see
-`PCIe atomics <https://pcisig.com/specifications/pciexpress/specifications/ECN_Atomic_Ops_080417.pdf>`_
-
-`Linux Kernel Patch to pci_enable_atomic_request <https://patchwork.kernel.org/project/linux-pci/patch/1443110390-4080-1-git-send-email-jay@jcornwall.me/>`_
-
-There are also a number of papers which talk about these new capabilities:
-
-  * `Atomic Read Modify Write Primitives by Intel <https://www.intel.es/content/dam/doc/white-paper/atomic-read-modify-write-primitives-i-o-devices-paper.pdf>`_
-  * `PCI express 3 Accelerator White paper by Intel <https://www.intel.sg/content/dam/doc/white-paper/pci-express3-accelerator-white-paper.pdf>`_
-  * `PCIe Generation 4 Base Specification includes atomic operations <https://astralvx.com/storage/2020/11/PCI_Express_Base_4.0_Rev0.3_February19-2014.pdf>`_
-  * `Xilinx PCIe Ultrascale White paper <https://docs.xilinx.com/v/u/8OZSA2V1b1LLU2rRCDVGQw>`_
-
-Other I/O devices with PCIe atomics support:
-
-  * Mellanox ConnectX-5 InfiniBand Card
-  * Cray Aries Interconnect
-  * Xilinx 7 Series Devices
+The atomic operations are initiated by the I/O devices that support 32-bit, 64-bit, and
+128-bit operand. The target address has to be naturally aligned to the operation size.
```

Comment:
```suggestion
128-bit operands. The target address has to be naturally aligned to the operation size.
```

"The target address has to be naturally aligned to the operation size." seems to come out of nowhere. It needs contextualization.

Reply:
Rewrote to provide more context.